MediaBrowser.Model/Cryptography/PasswordHash.cs

```diff
@@ -16,70 +16,78 @@ public class PasswordHash
+                        {
+                            Parameters.Add(fields[0], fields[1]);
+                        }
                     }
                 }
-                if (a.Length == 4)
+                if (SplitStorageString.Length == 5)
                 {
-                    Salt = a[2];
-                    SaltBytes = FromByteString(Salt);
-                    Hash = a[3];
-                    HashBytes = FromByteString(Hash);
+                    Salt = SplitStorageString[3];
+                    SaltBytes = ConvertFromByteString(Salt);
+                    Hash = SplitStorageString[4];
+                    HashBytes = ConvertFromByteString(Hash);
                 }
                 else
                 {
                     Salt = string.Empty;
-                    Hash = a[3];
-                    HashBytes = FromByteString(Hash);
+                    Hash = SplitStorageString[3];
+                    HashBytes = ConvertFromByteString(Hash);
                 }
             }
             else
             {
-                if (a.Length == 4)
+                if (SplitStorageString.Length == 4)
                 {
-                    Salt = a[2];
-                    SaltBytes = FromByteString(Salt);
-                    Hash = a[3];
-                    HashBytes = FromByteString(Hash);
+                    Salt = SplitStorageString[2];
+                    SaltBytes = ConvertFromByteString(Salt);
+                    Hash = SplitStorageString[3];
+                    HashBytes = ConvertFromByteString(Hash);
                 }
                 else
                 {
                     Salt = string.Empty;
-                    Hash = a[2];
-                    HashBytes = FromByteString(Hash);
+                    Hash = SplitStorageString[2];
+                    HashBytes = ConvertFromByteString(Hash);
                 }
 
             }
 
         }
 
-        public PasswordHash(ICryptoProvider cryptoProvider2)
+        public PasswordHash(ICryptoProvider cryptoProvider)
         {
-            Id = "SHA256";
-            SaltBytes = cryptoProvider2.GenerateSalt();
-            Salt = BitConverter.ToString(SaltBytes).Replace("-", "");
+            Id = cryptoProvider.DefaultHashMethod;
+            SaltBytes = cryptoProvider.GenerateSalt();
+            Salt = ConvertToByteString(SaltBytes);
         }
 
-        private byte[] FromByteString(string ByteString)
+        public static byte[] ConvertFromByteString(string byteString)
         {
             List<byte> Bytes = new List<byte>();
-            for (int i = 0; i < ByteString.Length; i += 2)
+            for (int i = 0; i < byteString.Length; i += 2)
             {
-                Bytes.Add(Convert.ToByte(ByteString.Substring(i, 2),16));
+                Bytes.Add(Convert.ToByte(byteString.Substring(i, 2),16));
             }
             return Bytes.ToArray();
-        }
+        }
+        public static string ConvertToByteString(byte[] bytes)
```

Comment:
add a delimiting empty line before the method pls

Reply:
done